### PR TITLE
[2175] Add editing of alert email address

### PR DESCRIPTION
**Pull request by timabell**
### Changes proposed in this pull request

Selection_112

Selection_120

### Guidance to review

### CODE REVIEW

**Comment by timabell** [comment]
[[M] Editable Email alerts for new applications](https://trello.com/c/57B30ZLF/2175-m-editable-email-alerts-for-new-applications)

File: app/controllers/ucas_contacts_controller.rb [hunk 1]
```diff
@@ -16,14 +16,23 @@ def show; end
   def alerts; end
 
   def update_alerts
-    @provider.update(provider_params)
-    redirect_to provider_ucas_contacts_path(@provider.provider_code),
-                flash: { success: "Your changes have been saved" }
+    email_changed = @provider.application_alert_contact != provider_params["application_alert_contact"]
+    permission_given = params["provider"]["share_with_ucas_permission"] != "1"
+    if permission_given && email_changed
+      @errors = { share_with_ucas_permission: ["Please give permission to share this email with UCAS"] }
```
Comment (fofr) [inline app/controllers/ucas_contacts_controller.rb]:
```suggestion
      @errors = { share_with_ucas_permission: ["Please give permission to share this email address with UCAS"] }
```

File: app/views/ucas_contacts/alerts.html.erb [hunk 2]
```diff
@@ -31,58 +32,52 @@
...
+        <%= render 'shared/error_messages', field: :share_with_ucas_permission %>
+        <div class="govuk-checkboxes" data-module="checkboxes">
+          <div class="govuk-checkboxes__item">
+            <%= form.check_box :share_with_ucas_permission,
+                                  checked: false,
+                                  class: 'govuk-checkboxes__input',
+                                  data: { qa: 'ucas_contacts__share_with_ucas_permission'}
+            %>
+            <%= form.label :share_with_ucas_permission,
+                           "I give permission to share this email with UCAS",
```
Comment (fofr) [inline app/views/ucas_contacts/alerts.html.erb]:
```suggestion
                           "I give permission to share this email address with UCAS",
```

**Comment by timabell** [comment]
Thanks for all the approval :+1: 
Just noticed my `<label>` tags aren't matched with the form elements

**Comment by timabell** [comment]
also @fofr tells me I'm missing the red border on the permission when not ticked :sob:

**Pull request merged**